selfhost/typechecker.jakt

```diff
@@ -7883,6 +7890,14 @@ struct Typechecker {
                                 )
                                 let checked_expression = .typecheck_expression_and_dereference_if_needed(new_condition, scope_id, safety_mode, type_hint: Some(subject_type_id), span)
 
+                                if is_boolean_match and checked_expression is Boolean(val) {
+                                    if val {
+                                        seen_true = true
+                                    } else {
+                                        seen_false = true
+                                    }
```

Comment:
```suggestion
                                    seen_true = val
```

Reply:
oh nevermind, I'm silly :P

Reply:
when I was first writing this I thought about this too! haha